amazonfreertossdk/src/main/java/software/amazon/freertos/amazonfreertossdk/AmazonFreeRTOSDevice.java

```diff
@@ -168,10 +169,12 @@ void disconnect() {
         mTotalPackets = 0;
         mPacketCount = 1;
 
-
+        /**
+         * Call disconnect but not close. Close will be called in the Gatt callback.
+         * This allows Android to track the changes
```

Comment:
what happens if GATT is already disconnected and user call disconnect ? Will the callback be invoked with an error ?

Reply:
I now moved the whole disconnect logic in the onConnectionStatusChange() callback